capstone/src/main/java/com/google/sps/servlets/ChallengeServlet.java

```diff
@@ -54,6 +84,24 @@ public void doPost(String userId, HttpServletRequest request, HttpServletRespons
             new ArrayList<String>() /* users completed */,
             0 /* id */);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(challenge.toEntity());
+    Entity challengeEntity = challenge.toEntity();
+    datastore.put(challengeEntity);
+    this.updateChallengesList(request, response, datastore, challengeEntity);
+  }
+
+  private void updateChallengesList(
+      HttpServletRequest request,
+      HttpServletResponse response,
+      DatastoreService datastore,
+      Entity challengeEntity)
+      throws IOException {
+    Entity groupEntity = Group.getGroupEntity(request, response, datastore);
+    List<Long> challenges =
+        (groupEntity.getProperty("challenges") == null)
+            ? new ArrayList<Long>()
+            : (List<Long>) groupEntity.getProperty("challenges");
+    challenges.add(challengeEntity.getKey().getId());
```

Comment:
[This is comment without any work required.] This order is only strictly maintained by the fact that all challenges CURRENTLY expire after the same amount of time, otherwise you may want to consider sorting by the `timestamp` property, especially since order-of-creation is easily checked by comparing IDs if you ever need to recreate that list.

Reply:
I think the list is ordered by when it is added into the database though